package/kedro_viz/integrations/kedro/data_loader.py

```diff
@@ -85,10 +87,15 @@ def load_data(
         A tuple containing the data catalog and the pipeline dictionary
         and the session store.
     """
-    from kedro.framework.project import pipelines
-    from kedro.framework.startup import bootstrap_project
+    from kedro.framework.project import configure_project, pipelines
+
+    if package_name:
+        configure_project(package_name)
+    else:
+        from kedro.framework.startup import bootstrap_project
 
-    bootstrap_project(project_path)
+        # bootstrap project when viz is run in dev mode
```

Comment:
What is dev mode and why this requires bootstrap_project?

Reply:
dev mode is when users do `make run` and does not use kedro cli. As of now we run the command - `python3 package/kedro_viz/server.py` during make run.

Reply:
hi, can we move this part to cli.py instead of here. So we can solve that problem that Juanlu raised #1761. Did the solution in this ticket - https://github.com/kedro-org/kedro-viz/pull/1793/files

Reply:
you are right, this needs some more refactoring to fix - #1761 .